File: android/app/src/main/java/com/nuvio/app/tv/ui/details/DetailsScreen.kt
package com.nuvio.app.tv.ui.details

import android.content.Context
import android.content.Intent
import android.content.res.Configuration
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalConfiguration
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import coil.compose.AsyncImage
import com.nuvio.app.tv.ui.components.*
import com.nuvio.app.tv.ui.theme.HeroGradient

@Composable
fun DetailsScreen(
    id: String,
    onPlayClick: (String) -> Unit,
    onBack: () -> Unit,
    viewModel: DetailsViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsState()
    val context = LocalContext.current
    val configuration = LocalConfiguration.current
    val isTv = configuration.uiMode and Configuration.UI_MODE_TYPE_MASK == Configuration.UI_MODE_TYPE_TELEVISION

    LaunchedEffect(id) {
        viewModel.loadDetails(id)
    }

    Box(modifier = Modifier.fillMaxSize()) {
        when {
            uiState.isLoading -> {
                CircularProgressIndicator(
                    modifier = Modifier.align(Alignment.Center)
                )
            }
            uiState.error != null -> {
                ErrorView(
                    error = uiState.error ?: "Unknown error",
                    onRetry = { viewModel.loadDetails(id) },
                    onBack = onBack,
                    modifier = Modifier.align(Alignment.Center)
                )
            }
            uiState.meta != null -> {
                if (isTv) {
                    TvDetailsContent(
                        uiState = uiState,
                        onPlayClick = {
                            uiState.streams.firstOrNull()?.url?.let { url ->
                                onPlayClick(url)
                            }
                        },
                        onWatchlistClick = { viewModel.toggleWatchlist() },
                        onRateClick = { /* TODO: Show rating dialog */ },
                        onShareClick = { shareContent(context, uiState.meta!!) },
                        onBack = onBack
                    )
                } else {
                    MobileDetailsContent(
                        uiState = uiState,
                        onPlayClick = {
                            uiState.streams.firstOrNull()?.url?.let { url ->
                                onPlayClick(url)
                            }
                        },
                        onWatchlistClick = { viewModel.toggleWatchlist() },
                        onRateClick = { /* TODO: Show rating dialog */ },
                        onShareClick = { shareContent(context, uiState.meta!!) },
                        onBack = onBack
                    )
                }
            }
        }
    }
}

@Composable
fun TvDetailsContent(
    uiState: DetailsUiState,
    onPlayClick: () -> Unit,
    onWatchlistClick: () -> Unit,
    onRateClick: () -> Unit,
    onShareClick: () -> Unit,
    onBack: () -> Unit
) {
    val meta = uiState.meta ?: return

    Box(modifier = Modifier.fillMaxSize()) {
        // Background image with gradient
        AsyncImage(
            model = meta.backgroundUrl ?: meta.posterUrl,
            contentDescription = null,
            contentScale = ContentScale.Crop,
            modifier = Modifier.fillMaxSize()
        )

        // Gradient overlay
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    Brush.verticalGradient(
                        colors = HeroGradient
                    )
                )
        )

        // Content
        LazyColumn(
            modifier = Modifier.fillMaxSize(),
            contentPadding = PaddingValues(48.dp)
        ) {
            item {
                // Back button
                IconButton(onClick = onBack) {
                    Icon(
                        imageVector = Icons.Default.ArrowBack,
                        contentDescription = "Back",
                        tint = Color.White
                    )
                }

                Spacer(modifier = Modifier.height(24.dp))
            }

            item {
                // Metadata info
                TvMetadataInfo(meta = meta)

                Spacer(modifier = Modifier.height(32.dp))
            }

            item {
                // Action buttons
                TvActionButtons(
                    onPlayClick = onPlayClick,
                    onWatchlistClick = onWatchlistClick,
                    onRateClick = onRateClick,
                    onShareClick = onShareClick,
                    isInWatchlist = uiState.isInWatchlist
                )

                Spacer(modifier = Modifier.height(48.dp))
            }

            item {
                // Cast and Crew
                CastCrewSection(
                    cast = meta.cast,
                    director = meta.director,
                    writer = meta.writer
                )
            }
        }
    }
}

@Composable
fun MobileDetailsContent(
    uiState: DetailsUiState,
    onPlayClick: () -> Unit,
    onWatchlistClick: () -> Unit,
    onRateClick: () -> Unit,
    onShareClick: () -> Unit,
    onBack: () -> Unit
) {
    val meta = uiState.meta ?: return

    Box(modifier = Modifier.fillMaxSize()) {
        // Background image with gradient
        AsyncImage(
            model = meta.backgroundUrl ?: meta.posterUrl,
            contentDescription = null,
            contentScale = ContentScale.Crop,
            modifier = Modifier
                .fillMaxWidth()
                .height(400.dp)
        )

        // Gradient overlay on background
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(400.dp)
                .background(
                    Brush.verticalGradient(
                        colors = listOf(
                            Color.Transparent,
                            Color(0x99000000),
                            MaterialTheme.colorScheme.background
                        )
                    )
                )
        )

        // Content
        LazyColumn(
            modifier = Modifier.fillMaxSize()
        ) {
            item {
                // Spacer for background image
                Spacer(modifier = Modifier.height(300.dp))
            }

            item {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(MaterialTheme.colorScheme.background)
                        .padding(24.dp)
                ) {
                    // Metadata info
                    MetadataInfo(meta = meta)

                    Spacer(modifier = Modifier.height(24.dp))

                    // Action buttons
                    ActionButtons(
                        onPlayClick = onPlayClick,
                        onWatchlistClick = onWatchlistClick,
                        onRateClick = onRateClick,
                        onShareClick = onShareClick,
                        isInWatchlist = uiState.isInWatchlist
                    )

                    Spacer(modifier = Modifier.height(32.dp))

                    // Cast and Crew
                    CastCrewSection(
                        cast = meta.cast,
                        director = meta.director,
                        writer = meta.writer
                    )

                    Spacer(modifier = Modifier.height(24.dp))
                }
            }
        }

        // Back button overlay
        IconButton(
            onClick = onBack,
            modifier = Modifier
                .padding(16.dp)
                .background(
                    color = Color.Black.copy(alpha = 0.5f),
                    shape = RoundedCornerShape(50)
                )
        ) {
            Icon(
                imageVector = Icons.Default.ArrowBack,
                contentDescription = "Back",
                tint = Color.White
            )
        }
    }
}

@Composable
fun ErrorView(
    error: String,
    onRetry: () -> Unit,
    onBack: () -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier.padding(32.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Text(
            text = "Error",
            style = MaterialTheme.typography.headlineMedium,
            color = MaterialTheme.colorScheme.error
        )

        Text(
            text = error,
            style = MaterialTheme.typography.bodyLarge,
            color = MaterialTheme.colorScheme.onSurface
        )

        Row(
            horizontalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Button(onClick = onRetry) {
                Text("Retry")
            }

            OutlinedButton(onClick = onBack) {
                Text("Go Back")
            }
        }
    }
}

private fun shareContent(context: Context, meta: com.nuvio.app.tv.data.repository.Meta) {
    val shareText = buildString {
        append("Check out ${meta.name}")
        meta.year?.let { append(" ($it)") }
        append("\n\n")
        meta.description?.let { append(it) }
        meta.imdbId?.let { append("\n\nhttps://www.imdb.com/title/$it") }
    }

    val intent = Intent(Intent.ACTION_SEND).apply {
        type = "text/plain"
        putExtra(Intent.EXTRA_TEXT, shareText)
        putExtra(Intent.EXTRA_SUBJECT, meta.name)
    }

    context.startActivity(Intent.createChooser(intent, "Share ${meta.name}"))
}

File: ios/NuvioTV/Sources/UI/Details/DetailsScreen.swift
//
//  DetailsScreen.swift
//  NuvioTV
//
//  Created by Claude Code
//  Content details screen with adaptive layouts for iOS/iPad/tvOS
//

import SwiftUI

struct DetailsScreen: View {
    let id: String
    let onPlayClick: (String) -> Void
    let onBack: () -> Void

    @StateObject private var viewModel: DetailsViewModel
    @Environment(\.horizontalSizeClass) private var horizontalSizeClass

    init(
        id: String,
        repository: CatalogRepository,
        onPlayClick: @escaping (String) -> Void,
        onBack: @escaping () -> Void
    ) {
        self.id = id
        self.onPlayClick = onPlayClick
        self.onBack = onBack
        _viewModel = StateObject(wrappedValue: DetailsViewModel(repository: repository))
    }

    var body: some View {
        ZStack {
            if viewModel.uiState.isLoading {
                ProgressView()
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            } else if let error = viewModel.uiState.error {
                ErrorView(
                    error: error,
                    onRetry: { viewModel.loadDetails(id: id) },
                    onBack: onBack
                )
            } else if viewModel.uiState.meta != nil {
                #if os(tvOS)
                TvDetailsContent(
                    uiState: viewModel.uiState,
                    onPlayClick: {
                        if let url = viewModel.uiState.streams.first?.url {
                            onPlayClick(url)
                        }
                    },
                    onWatchlistClick: { viewModel.toggleWatchlist() },
                    onRateClick: { /* TODO: Show rating dialog */ },
                    onShareClick: { shareContent(viewModel.uiState.meta!) },
                    onBack: onBack
                )
                #else
                MobileDetailsContent(
                    uiState: viewModel.uiState,
                    onPlayClick: {
                        if let url = viewModel.uiState.streams.first?.url {
                            onPlayClick(url)
                        }
                    },
                    onWatchlistClick: { viewModel.toggleWatchlist() },
                    onRateClick: { /* TODO: Show rating dialog */ },
                    onShareClick: { shareContent(viewModel.uiState.meta!) },
                    onBack: onBack
                )
                #endif
            }
        }
        .onAppear {
            viewModel.loadDetails(id: id)
        }
    }

    private func shareContent(_ meta: Meta) {
        var shareText = "Check out \(meta.name)"
        if let year = meta.year {
            shareText += " (\(year))"
        }
        shareText += "\n\n"
        if let description = meta.description {
            shareText += description
        }
        if let imdbId = meta.imdbId {
            shareText += "\n\nhttps://www.imdb.com/title/\(imdbId)"
        }

        #if !os(tvOS)
        let activityVC = UIActivityViewController(
            activityItems: [shareText],
            applicationActivities: nil
        )

        if let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
           let rootVC = windowScene.windows.first?.rootViewController {
            rootVC.present(activityVC, animated: true)
        }
        #endif
    }
}

struct TvDetailsContent: View {
    let uiState: DetailsUiState
    let onPlayClick: () -> Void
    let onWatchlistClick: () -> Void
    let onRateClick: () -> Void
    let onShareClick: () -> Void
    let onBack: () -> Void

    var body: some View {
        guard let meta = uiState.meta else { return AnyView(EmptyView()) }

        return AnyView(
            ZStack {
                // Background image with gradient
                if let backgroundUrl = meta.backgroundUrl ?? meta.posterUrl {
                    AsyncImage(url: URL(string: backgroundUrl)) { image in
                        image
                            .resizable()
                            .aspectRatio(contentMode: .fill)
                    } placeholder: {
                        Color.black
                    }
                    .ignoresSafeArea()
                }

                // Gradient overlay
                LinearGradient(
                    colors: [
                        Color.black.opacity(0.3),
                        Color.black.opacity(0.7),
                        Color.black
                    ],
                    startPoint: .top,
                    endPoint: .bottom
                )
                .ignoresSafeArea()

                // Content
                ScrollView {
                    VStack(alignment: .leading, spacing: 48) {
                        // Back button
                        Button(action: onBack) {
                            Image(systemName: "chevron.left")
                                .font(.title)
                                .foregroundColor(.white)
                        }
                        .buttonStyle(.plain)

                        // Metadata info
                        TvMetadataInfo(meta: meta)

                        // Action buttons
                        TvActionButtons(
                            onPlayClick: onPlayClick,
                            onWatchlistClick: onWatchlistClick,
                            onRateClick: onRateClick,
                            onShareClick: onShareClick,
                            isInWatchlist: uiState.isInWatchlist
                        )

                        // Cast and Crew
                        CastCrewSection(
                            cast: meta.cast,
                            director: meta.director,
                            writer: meta.writer
                        )
                    }
                    .padding(48)
                }
            }
        )
    }
}

struct MobileDetailsContent: View {
    let uiState: DetailsUiState
    let onPlayClick: () -> Void
    let onWatchlistClick: () -> Void
    let onRateClick: () -> Void
    let onShareClick: () -> Void
    let onBack: () -> Void

    var body: some View {
        guard let meta = uiState.meta else { return AnyView(EmptyView()) }

        return AnyView(
            ZStack(alignment: .top) {
                ScrollView {
                    VStack(spacing: 0) {
                        // Background image with gradient
                        ZStack(alignment: .bottom) {
                            if let backgroundUrl = meta.backgroundUrl ?? meta.posterUrl {
                                AsyncImage(url: URL(string: backgroundUrl)) { image in
                                    image
                                        .resizable()
                                        .aspectRatio(contentMode: .fill)
                                } placeholder: {
                                    Color.black
                                }
                                .frame(height: 400)
                                .clipped()
                            }

                            // Gradient overlay
                            LinearGradient(
                                colors: [
                                    Color.clear,
                                    Color.black.opacity(0.6),
                                    Color.black
                                ],
                                startPoint: .top,
                                endPoint: .bottom
                            )
                            .frame(height: 400)
                        }

                        // Content
                        VStack(alignment: .leading, spacing: 24) {
                            // Metadata info
                            MetadataInfo(meta: meta)

                            // Action buttons
                            ActionButtons(
                                onPlayClick: onPlayClick,
                                onWatchlistClick: onWatchlistClick,
                                onRateClick: onRateClick,
                                onShareClick: onShareClick,
                                isInWatchlist: uiState.isInWatchlist
                            )

                            // Cast and Crew
                            CastCrewSection(
                                cast: meta.cast,
                                director: meta.director,
                                writer: meta.writer
                            )
                        }
                        .padding(24)
                        .background(Color.black)
                    }
                }
                .ignoresSafeArea(edges: .top)

                // Back button overlay
                Button(action: onBack) {
                    Image(systemName: "chevron.left")
                        .font(.title2)
                        .foregroundColor(.white)
                        .frame(width: 44, height: 44)
                        .background(
                            Circle()
                                .fill(Color.black.opacity(0.5))
                        )
                }
                .buttonStyle(.plain)
                .padding(16)
            }
        )
    }
}

struct ErrorView: View {
    let error: String
    let onRetry: () -> Void
    let onBack: () -> Void

    var body: some View {
        VStack(spacing: 16) {
            Text("Error")
                .font(.title)
                .foregroundColor(.red)

            Text(error)
                .font(.body)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)

            HStack(spacing: 16) {
                Button("Retry", action: onRetry)
                    .buttonStyle(.borderedProminent)

                Button("Go Back", action: onBack)
                    .buttonStyle(.bordered)
            }
        }
        .padding(32)
    }
}
